### ISSUE TITLE
🐛 BUG: 3rd party renderer issues

### ISSUE THREAD

**Issue by trashhalo** [snowpackjs/astro#1164]
### Background & Motivation

#1097 problems hit while implementing rfc as a 3rd party renderer as requested

* Test case needs access to /dist/build, runtime and config. These are hidden by package.json exports. Currently working around this by patching astro
* Create runtime can only load renderers in "external" packages. It takes the name of the renderer and slaps the server/client on the end and trys to resolve it. This works in astro due to lerna workspaces. I'm working around this by setting up a lerna workspace... For my one package. Then putting the tests in one package and the renderer in another. 

### Possible solutions
There's not really a issue type for this kind of thing so I'm typing something here to make this happy.

### Help make it happen!

- [ ] I am willing to submit a PR to implement this change.
- [ ] I am willing to submit a PR to implement this change, but would need some guidance.
- [ ] I am not willing to submit a PR to implement this change.

**Comment by trashhalo** [comment]
https://github.com/trashhalo/astrojs-renderer-react-next

**Comment by FredKSchott** [comment]
@trashhalo this would be considered a bug, since we definitely have buy-in to support 3rd party renderers.

**Comment by trashhalo** [comment]
@FredKSchott Saw this ticket got marked blocked. Anything I can help with?

**Comment by FredKSchott** [comment]
I think that just meant "blocked by new compiler", since all `compiler-v2` issues were moved into blocked. You can still take a look at this if you'd like, but the core team is focusing on getting the new compiler merged first and then will help prioritize this in the new compiler (if its even still an issue).

**Comment by natemoo-re** [comment]
Update! The new compiler and build engine (backed by Vite) have been merged to `main` and are in preview under `astro@next`! We still have to write up some docs on how to upgrade your renderers, but you can take a look at [the existing React renderer](https://github.com/snowpackjs/astro/tree/2efee75989411474f02890ffb86cc2fb62e42b4e/packages/renderers/renderer-react) for now.

We've also done a lot of work to refactor our tests. I _think_ that first issue should be fixed now, but we still don't have a great test harness for Astro or anything.

As for the `external` issue, we're not seeing any issues like that with Vite. We've also given renderer authors (almost) full control over the vite config in case they need to mark particular packages as external. I'm confident that won't be a problem when you upgrade!

Really appreciate your work on this, it's a very cool idea!

**Comment by natemoo-re** [comment]
Following up on my previous comment, the team has decided to close out issues that have been addressed by [astro@0.21.0-next.0](https://github.com/snowpackjs/astro/releases/tag/astro%400.21.0-next.0), [astro@0.21.0-next.1](https://github.com/snowpackjs/astro/releases/tag/astro%400.21.0-next.1), or [astro@0.21.0-next.2](https://github.com/snowpackjs/astro/releases/tag/astro%400.21.0-next.2). Our hope is that this will help the [v0.21 milestone](https://github.com/snowpackjs/astro/milestone/1) remain as actionable as possible.

Our new documentation on the Renderer APIs will go out in tandem with the promotion of `astro@next` to `astro`.

If you think we made a mistake, **please reopen this issue**!

If you run into any other problems with `astro@next`, please [open a new issue](https://github.com/snowpackjs/astro/issues/new?assignees=&labels=0.21&template=---01-bug-report.yml&title=%F0%9F%90%9B+BUG%3Aastro@next).
